Desafio_Snowman/use_cases_turismo.py

```diff
@@ -30,6 +28,26 @@ def haversine(lon1, lat1, lon2, lat2):  # def que aplica a formula de haversine
     r = 6371
     return c * r
 
+def token_required(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        data = request.json
+        access_token = request.headers.get("Authorization") #chega o token com bearer precedendo ele
+
+        if not(access_token):
+            return jsonify({'message': 'token is missing!'}),401
+
+        access_token = access_token.split(" ")[1] #removo o bearer com o split
+        try:
+            auth.Token().decode_json_web_token(access_token) #chamo minha função decode que retorna o email contido no token
```

Comment:
Sorry, i can't see... I only generate the token in the function **login_logica()** if the datas informed by user are valid in the function **validar_email_senha_do_usuario()**. So, if the token exists, it is automatically valid and the user exists.
  Considering this, why should I check it again?

Reply:
I guess if the user want to delete your account,  checking if the user exists in this part is very good idea, because he need a token to access a future account deletion function and, after deleting account, the token needs to be invalited.

Reply:
It's just to guarantee the user is active yet. He could have ben deleted. In some cases you would check if it is active, and this sort of thing.